cmd/kube-scheduler/app/server.go

```diff
@@ -180,9 +180,13 @@ func Run(ctx context.Context, cc *schedulerserverconfig.CompletedConfig, sched *
 
 	// Start all informers.
 	cc.InformerFactory.Start(ctx.Done())
-
 	// Wait for all caches to sync before scheduling.
 	cc.InformerFactory.WaitForCacheSync(ctx.Done())
+	// DynInformerFactory can be nil in tests.
+	if cc.DynInformerFactory != nil {
+		cc.DynInformerFactory.Start(ctx.Done())
```

Comment:
We should do both `Start`s first and then `WaitForCacheSync`. Otherwise we are serializing the syncs.

Here and everywhere else.

Reply:
I don't get it. It should suffice to ensure for each informerFactory, its Start() occurs before its WaitForCacheSync(). In other words, what's the side effect of starting dynInformerFactory after regular informerFactory's sync?

Reply:
My recommendation is not about order guarantees. What you have works.

However, it might be slightly faster if you first start all the workers and then wait for the sync of both. Rather than start the one informer, wait for the sync and then start the second informer.

Reply:
> it might be slightly faster if you first start all the workers and then wait for the sync of both

That sounds fair.

BTW: I will append a commit, and recently I learned we can label a PR with command `/label tide/merge-method-squash`, so it will be auto-squashed on the fly and then merged into the git tree (but in the Github GUI, the PR still shows 2 commits). Let's give it a try.